lang: python: readme: document mechanism for corner-case python[3]-package.mk location

commodo opened the pull request:
Maintainer: me, @jefferyto 
Compile tested: n/a
Run tested: n/a

-------------------------------------------------------------------

After some thinking over this, documenting this behavior makes sense
versus adding some functionst to handle this.
There is some validity/use-cases where some users may want to reference
a python[3]-package.mk from some other location as well as have the
flexibility to change it (locally). One example can be when the local
`packages` is renamed to something else.

This does not fall on the responsibility of the Python maintainers, but
it can be documented.

Signed-off-by: Alexandru Ardelean <ardeleanalex@gmail.com>

commodo commented:
@pmelange
notifying you;
see if this is ok;

jefferyto commented on lang/python/README.md:
````diff
@@ -103,6 +103,26 @@ include $(TOPDIR)/feeds/openwrt-packages/lang/python/python3-package.mk
+In case there is a need/requirement such that the local package feed is named something else than `packages`, one approach to make the package flexible to change is:
+
+```
+PYTHON_PACKAGE_MK:=$(wildcard $(TOPDIR)/feeds/*/python-package.mk)
````
Does the `*` search subdirectories or would this need to be `PYTHON_PACKAGE_MK:=$(wildcard $(TOPDIR)/feeds/*/lang/python/python-package.mk)`?

jefferyto commented on lang/python/README.md:
````diff
@@ -103,6 +103,26 @@ include $(TOPDIR)/feeds/openwrt-packages/lang/python/python3-package.mk
+else
+$(info found python-package.mk at $(PYTHON_PACKAGE_MK))
+endif
+```
````
Maybe it would good to explicitly include
```
include $(PYTHON_PACKAGE_MK)
```
here (to avoid confusion :laughing:)?

pmelange commented:
I agree with @jefferyto.

commodo commented:
Changelog v1 -> v2:
* updated `PYTHON_PACKAGE_MK:=$(wildcard $(TOPDIR)/feeds/*/python-package.mk)` -> `PYTHON_PACKAGE_MK:=$(wildcard $(TOPDIR)/feeds/*/lang/python/python-package.mk)`
* added `include $(PYTHON_PACKAGE_MK)`

pmelange commented:
looks good to me

commodo commented on lang/python/README.md:
````diff
@@ -103,6 +103,26 @@ include $(TOPDIR)/feeds/openwrt-packages/lang/python/python3-package.mk
+In case there is a need/requirement such that the local package feed is named something else than `packages`, one approach to make the package flexible to change is:
+
+```
+PYTHON_PACKAGE_MK:=$(wildcard $(TOPDIR)/feeds/*/python-package.mk)
````
i'm confused;
isn't it like that ?
i.e.  
```
PYTHON_PACKAGE_MK:=$(wildcard $(TOPDIR)/feeds/*/lang/python/python-package.mk)
```
?

to me it looks like it is

pmelange replied:
For some reason my review from Monday is just showing up now.  Ignore it (and I'm deleting it).
